File: ts/management-api-v2/cm_api_v2_post_taxonomy_group.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client.addTaxonomy()
  .withData(
    {
      name: 'Personas',
      externalId: 'Tax-Group-123',
      codename: 'people',
      terms: [
        {
          name: 'Coffee expert',
          externalId: 'Tax-term-456',
          codename: 'expert',
          terms: [
            {
              name: 'Barista',
              externalId: 'Tax-term-789',
              terms: []
            },
            {
              name: 'Cafe owner',
              externalId: 'Tax-term-101',
              terms: []
            }
          ]
        },
        {
          name: 'Coffee enthusiast',
          codename: 'enthusiast',
          externalId: 'Tax-term-112',
          terms: [
            {
              name: 'Coffee lover',
              codename: 'lover',
              externalId: 'Tax-term-131',
              terms: []
            },
            {
              name: 'Coffee blogger',
              codename: 'blogger',
              externalId: 'Tax-term-145',
              terms: []
            }
          ]
        }
      ]
    }
  )
  .toPromise();

File: net/management-api-v2/cm_api_v2_post_taxonomy_group.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = await client.CreateTaxonomyGroupAsync(new TaxonomyGroupCreateModel
{
    Name = "Personas",
    ExternalId = "Tax-Group-123",
    Codename = "people",
    Terms = new TaxonomyTermCreateModel[]
    {
        new TaxonomyTermCreateModel
        {
            Name = "Coffee expert",
            Codename = "expert",
            ExternalId = "Tax-term-456",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Barista",
                    ExternalId = "Tax-term-789",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Cafe owner",
                    ExternalId = "Tax-term-101",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                }
            }
        },
        new TaxonomyTermCreateModel
        {
            Name = "Coffee enthusiast",
            Codename = "enthusiast",
            ExternalId = "Tax-term-112",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Coffee lover",
                    ExternalId = "Tax-term-131",
                    Codename = "lover",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Coffee blogger",
                    ExternalId = "Tax-term-145",
                    Codename = "blogger",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                }
            }
        }
    }
});
